operator/cmd/root.go

```diff
@@ -663,13 +664,29 @@ func (legacy *legacyOnLeader) onStart(_ hive.HookContext) error {
 	}
 
 	if legacy.clientset.IsEnabled() {
-		if operatorOption.Config.EndpointGCInterval != 0 {
+		// Conditionally start the CiliumEndpoint garbage collector.
+		// The GC needs to continually run long-term if EndpointGCInterval is non-zero and if CiliumEndpoint CRDs
+		// are enabled. Otherwise, the GC still needs to run once to account for the case in which a user transitions
+		// from CiliumEndpoint CRD mode to kvstore mode, and to check if there are any stale CEPs that need to be
+		// purged.
+		if operatorOption.Config.EndpointGCInterval != 0 && !option.Config.DisableCiliumEndpointCRD {
 			enableCiliumEndpointSyncGC(legacy.ctx, &legacy.wg, legacy.clientset, ciliumNodeSynchronizer, false)
 		} else {
-			// Even if the EndpointGC is disabled we still want it to run at least
-			// once. This is to prevent leftover CEPs from populating ipcache with
-			// stale entries.
-			enableCiliumEndpointSyncGC(legacy.ctx, &legacy.wg, legacy.clientset, ciliumNodeSynchronizer, true)
+			// Check if CEP CRD is available, as it could be missing if CEPs are not enabled.
+			// If the CRD is not available, then no garbage collection needs to be done.
+			_, err := legacy.clientset.ApiextensionsV1().CustomResourceDefinitions().Get(
+				legacy.ctx, k8sconstv2.CEPName, metav1.GetOptions{},
```

Comment:
Sorry, one minor nit. Similar to what we are doing in https://github.com/cilium/cilium/blob/8f9af4663ac75d5fa2e79200bd66625bf12b8639/pkg/k8s/watchers/endpointsynchronizer.go#L144-L147

We should do a get `meta_v1.GetOptions{ResourceVersion: "0"})`

Reply:
Good catch, fixed